sqlglot/dialects/tsql.py

```diff
@@ -518,6 +514,36 @@ def _parse_user_defined_function(
             expressions = self._parse_csv(self._parse_function_parameter)
             return self.expression(exp.UserDefinedFunction, this=this, expressions=expressions)
 
+        def _parse_id_var(
+            self,
+            any_token: bool = True,
+            tokens: t.Optional[t.Collection[TokenType]] = None,
+        ) -> t.Optional[exp.Expression]:
```

Comment:
is_temporary = self._match(TokenType.Hash)
is_global = is_temporary and  self._match(TokenType.Hash)